### PR TITLE
Apply monitoring linters

### PR DESCRIPTION
**Pull request by assafad**
**What this PR does / why we need it**:
Apply the following linters:
- [monitoringlinter](https://github.com/kubevirt/monitoring/tree/main/monitoringlinter) that was implemented in https://github.com/kubevirt/monitoring/pull/221, and was designed to enforce https://github.com/kubevirt/community/pull/219.
This linter ensures that all monitoring-related practices are implemented within the `pkg/monitoring` directory using [operator-observability](https://github.com/machadovilaca/operator-observability/tree/main) methods. It verifies that all metrics, alerts and recording rules registrations are centralized in this directory, and restricts the direct use of [Prometheus registration methods](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#Registerer).
Example for a message reported by the linter:

  ```
  kubevirt/pkg/virt-controller/watch/workload-updater/workload-updater.go:67:2: monitoring-linter: metrics should be registered only within pkg/monitoring directory, using operator-observability packages.
  ```

- [operator-observability rules linter](https://github.com/machadovilaca/operator-observability/tree/main/pkg/testutil) - ensures that alerts and recording rules definitions are following the monitoring best practices, e.g. alerts have all the required annotations and labels with expected values, recording rules have expressions, etc.

fixes https://issues.redhat.com/browse/CNV-36765
Other PR reference: https://github.com/kubevirt/hyperconverged-cluster-operator/pull/2827

**Release note**:

```release-note
None
```

### CODE REVIEW

**Comment by assafad** [comment]
/hold
blocked by Blocked by https://issues.redhat.com/browse/CNV-36845.

File: hack/build/run-linters.sh [hunk 1]
```diff
@@ -1,7 +1,10 @@
 #!/bin/sh -ex
 
 GOLANGCI_VERSION="${GOLANGCI_VERSION:-v1.58.0}"
+MONITORING_LINTER_VERSION="${MONITORING_LINTER_VERSION:-e2be790}"
 
 go install "github.com/golangci/golangci-lint/cmd/golangci-lint@${GOLANGCI_VERSION}"
+go install "github.com/kubevirt/monitoring/monitoringlinter/cmd/monitoringlinter@${MONITORING_LINTER_VERSION}"
 
 golangci-lint run --timeout=16m ./... "$@"
+monitoringlinter ./...
```
Comment (machadovilaca) [inline hack/build/run-linters.sh:10]: nit: add empty line at the end of file

**Comment by machadovilaca** [comment]
/lgtm
just left a small nit

File: hack/build/run-linters.sh (hunk 1 above)
Comment (assafad) [reply hack/build/run-linters.sh]: done, thanks

**Pull request merged**